### PR TITLE
use Index_t instead of index_t

### PR DESCRIPTION
**Pull request by FrankYu**
on smartos system(open solaris system like), index_t is used in file
sys/types.h, so modify index_t to Index_t to skip the conflict.

Signed-off-by: Frank Yu <flyxiaoyu@gmail.com>

### CODE REVIEW

**Comment by FrankYu** [comment]
this fix the issue https://github.com/PyWavelets/pywt/issues/177

**Comment by kwohlfahrt** [comment]
`index_t` is not [POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html) for `sys/types.h`. I'm not sure we want to support this, but the others should weigh in on that too.

Apart from that:

- Is there any way to set up CI testing for SmartOS? Otherwise I'd be reluctant to change things to 
support it if we cannot reliably test it.
- Secondly, I would use explicit namespacing - `pywt_index_t`, because `Index_t` doesn't give much insight into why it is uppercase when other primitive types are not.

**Comment by FrankYu** [comment]
@kwohlfahrt 
1) I don't know how to setup CI testing for SmartOS, if anything I can do I'd like to do this.
2) agree with you, I have modify Index_t to pywt_index_t

Thanks

**Comment by rgommers** [comment]
> Is there any way to set up CI testing for SmartOS? Otherwise I'd be reluctant to change things to support it if we cannot reliably test it.

Extra CI is always nice, but anyway I think the change to ``pywt_index_t`` is a healthy one - so no reason not to merge this PR.

**Comment by FrankYu** [comment]
@rgommers 
thanks:)

File: pywt/_extensions/c/common.h [hunk 1]
```diff
@@ -23,14 +23,14 @@
         #include "Python.h"
     #endif
 
-    typedef Py_ssize_t index_t;
+    typedef Py_ssize_t pywt_index_t;
```
Comment (kwohlfahrt) [inline pywt/_extensions/c/common.h]: If we won't have CI then put a comment in here mentioning SmartOS/Solaris at the reason for namespacing.

**Comment by kwohlfahrt** [comment]
+1 from me then.

**Comment by rgommers** [comment]
Allright, in it goes then. Thanks @FrankYu, @kwohlfahrt

**Pull request merged**